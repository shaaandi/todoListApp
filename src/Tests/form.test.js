import React from 'react';
import {shallow} from 'enzyme';
import Form from '../components/jsfiles/form';

let wrapper = shallow(<Form addTodo={() => {}}/>);

describe('Form Testing ', () => {
    it('Form SnapShot Testing', () => {
        expect(wrapper).toMatchSnapshot();
    })


    it('Testing the Submit Handler and Change Handler', () => {
        let value = 'Drink Coffee'
        let title = 'title'
        let event = {
            target : {
               value  : value ,
               name : title
            },
            preventDefault : () => {}
        }
        wrapper.instance().changeHandler(event)
        expect(wrapper.state()).toEqual({title: value, flag : 'white' })
        wrapper.instance().submitHandler(event)
        expect(wrapper.state()).toEqual({title: '', flag : 'white' })
        
    })
})