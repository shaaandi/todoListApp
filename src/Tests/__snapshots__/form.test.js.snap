// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Testing  Form SnapShot Testing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form
    addTodo={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          id="#test"
          name="title"
          onChange={[Function]}
          placeholder="+ Add new Todo"
          type="text"
          value=""
        />,
        <select
          className="flagChooser"
          name="flag"
          onChange={[Function]}
          value="white"
        >
          <option
            value="white"
          >
            White
          </option>
          <option
            value="green"
          >
            Green
          </option>
          <option
            value="red"
          >
            Red
          </option>
        </select>,
      ],
      "className": "junk",
      "id": "form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "#test",
          "name": "title",
          "onChange": [Function],
          "placeholder": "+ Add new Todo",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value="white"
            >
              White
            </option>,
            <option
              value="green"
            >
              Green
            </option>,
            <option
              value="red"
            >
              Red
            </option>,
          ],
          "className": "flagChooser",
          "name": "flag",
          "onChange": [Function],
          "value": "white",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "White",
              "value": "white",
            },
            "ref": null,
            "rendered": "White",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Green",
              "value": "green",
            },
            "ref": null,
            "rendered": "Green",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Red",
              "value": "red",
            },
            "ref": null,
            "rendered": "Red",
            "type": "option",
          },
        ],
        "type": "select",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            id="#test"
            name="title"
            onChange={[Function]}
            placeholder="+ Add new Todo"
            type="text"
            value=""
          />,
          <select
            className="flagChooser"
            name="flag"
            onChange={[Function]}
            value="white"
          >
            <option
              value="white"
            >
              White
            </option>
            <option
              value="green"
            >
              Green
            </option>
            <option
              value="red"
            >
              Red
            </option>
          </select>,
        ],
        "className": "junk",
        "id": "form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "#test",
            "name": "title",
            "onChange": [Function],
            "placeholder": "+ Add new Todo",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                value="white"
              >
                White
              </option>,
              <option
                value="green"
              >
                Green
              </option>,
              <option
                value="red"
              >
                Red
              </option>,
            ],
            "className": "flagChooser",
            "name": "flag",
            "onChange": [Function],
            "value": "white",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "White",
                "value": "white",
              },
              "ref": null,
              "rendered": "White",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Green",
                "value": "green",
              },
              "ref": null,
              "rendered": "Green",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Red",
                "value": "red",
              },
              "ref": null,
              "rendered": "Red",
              "type": "option",
            },
          ],
          "type": "select",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
